tests/backup_helper.go

```diff
@@ -4920,6 +4948,41 @@ func CreateRuleForBackupWithMultipleApplications(orgID string, appList []string,
 	return preRuleName, postRuleName, nil
 }
 
+type VMRuleType string
+
+const (
+	Freeze   VMRuleType = "freeze"
+	Unfreeze VMRuleType = "unfreeze"
+)
+
+func CreateRuleForVMBackup(ruleName string, vms []kubevirtv1.VirtualMachine, ruleType VMRuleType, ctx context1.Context) error {
+	var rulesInfo api.RulesInfo
+	namespaceToVMs := GetNamespacesToVMsMap(vms)
+	for namespace, vmList := range namespaceToVMs {
+		for _, vm := range vmList {
+			freezeAction := fmt.Sprintf("/usr/bin/virt-freezer --%s --name %s --namespace %s", ruleType, vm, namespace)
+			ruleAction := api.RulesInfo_Action{Background: false, RunInSinglePod: false,
+				Value: freezeAction}
+			var actions = []*api.RulesInfo_Action{&ruleAction}
+
+			rulesInfo.Rules = append(rulesInfo.Rules, &api.RulesInfo_RuleItem{
+				PodSelector: map[string]string{"vm.kubevirt.io/name": vm},
+				Actions:     actions,
+				Container:   VirtLauncherContainerName,
+			})
+		}
+	}
+	RuleCreateReq := &api.RuleCreateRequest{
```

Comment:
Can't we use the existing createRule function?

Reply:
The create rule function that we have today is app specific, but he VM rule is not specific to any VM. Every VM will have the same rule. Only the VM name and namespace will change.

Reply:
@mkoppal-px can we move this to a function and call it at two places?

Reply:
Did not get you

Reply:
We can take this up later